test/unit/rules.spec.ts

```diff
@@ -0,0 +1,51 @@
+import {expect} from 'chai';
+import {
+  ResourceGraphRule,
+  ResourceGraphExecutor,
+  RuleType,
+} from '../../src/rules';
+import {ResourceGraphModels} from '@azure/arm-resourcegraph';
+import {HttpHeadersLike, WebResourceLike} from '@azure/ms-rest-js';
+
+describe('Resource Graph Rule', () => {
+  const mockResourcesResponse: ResourceGraphModels.ResourcesResponse = {
+    totalRecords: 1,
+    count: 1,
+    resultTruncated: 'false',
+    data: {columns: [{name: 'id', type: 'string'}], rows: [['mockResourceId']]},
+    _response: {
+      request: {} as WebResourceLike,
+      status: 200,
+      headers: {} as HttpHeadersLike,
+      bodyAsText: '',
+      parsedBody: {} as ResourceGraphModels.QueryResponse,
+    },
+  };
```

Comment:
is there a preferred approach to mocking something like this?

Reply:
A const object like this works great. If we need to reuse across files, pulling this out into a `testData.ts` file works as well. There are also tools that can run/record HTTP requests, save to disk, & then replay with that saved data.